dhcpv4/nclient4/client.go

```diff
@@ -76,58 +77,39 @@ type pendingCh struct {
 
 // Logger is a handler which will be used to output logging messages
 type Logger interface {
-	// PrintMessage print _all_ DHCP messages
-	PrintMessage(prefix string, message *dhcpv4.DHCPv4)
+	ulog.Logger
```

Comment:
In my extremely humble opinion it is wrong to require `Print(v ...interface{})` if you do not use it and do not plan to use it.